src/Tools/dotnet-monitor/Auth/GeneratedApiKey.cs

```diff
@@ -23,15 +27,25 @@ private GeneratedApiKey(string monitorApiKey, string hashAlgorithm, string hashV
 
         public static GeneratedApiKey Create()
         {
-            return GeneratedApiKey.Create("SHA256");
+            return GeneratedApiKey.Create(DefaultKeyLength, DefaultHashAlgorithm);
         }
 
-        private static GeneratedApiKey Create(string hashAlgorithm)
+        public static GeneratedApiKey Create(int keyLength, string hashAlgorithm)
         {
+            if (!HashAlgorithmChecker.IsAllowedAlgorithm(hashAlgorithm))
```

Comment:
Are these exceptions visible in the logs after configuration validation? If so, maybe they can be more verbose?